forem/forem
# Remove bootstrap from admin/pages & admin/profile_fields and use Crayons

## What type of PR is this? (check all applicable)


## Description

This PR is a step towards removing bootstrap completely and replacing it with crayons.

The changes are only design related for which all before vs after screenshots have been shown below.

## Related Tickets & Documents



- Related Issue #18756
- Closes #

## QA Instructions, Screenshots, Recordings

### Pages

**How to test?**
1. Sign in as admin locally.
2. Visit `/admin/customization/config` and at the end expand `User Experience and Brand`
3. Uncheck the `Public` checkbox and update seetings.
4. Now visit `/admin/customization/pages`
5. Click `New Page`, fill out the details and when you click `Use as Locked Screen` this will trigger a modal for which the bootstrap was removed. (See screenshots below)

| Before  | After |
| ------------- | ------------- |
| <img width="1512" alt="Screenshot 2022-12-18 at 1 29 57 PM" src="https://user-images.githubusercontent.com/9396084/208293616-b58bd340-1b83-42d6-ab5b-d5eae27fa81f.png"> | <img width="1510" alt="Screenshot 2022-12-18 at 1 30 23 PM" src="https://user-images.githubusercontent.com/9396084/208293625-02c78436-adb1-4245-b234-104bbfcd3884.png"> |

### Profile Fields

**How to test?**
1. Sign in as admin locally and visit `/admin/customization/profile_fields`
2. Here you can experiment with values options and fields.

| Before  | After |
| ------------- | ------------- |
| <img width="1512" alt="Screenshot 2022-12-18 at 3 00 38 PM" src="https://user-images.githubusercontent.com/9396084/208293709-41883c51-4af8-4e9d-97e0-a895abab18de.png"> | <img width="1512" alt="Screenshot 2022-12-18 at 3 03 52 PM" src="https://user-images.githubusercontent.com/9396084/208293714-231649c1-1252-472d-925e-4b38a865c1cc.png"> |
| <img width="1512" alt="Screenshot 2022-12-18 at 3 04 15 PM" src="https://user-images.githubusercontent.com/9396084/208293716-5c3a2dcd-4d88-4d2b-bc94-6494ba88770a.png"> | <img width="1512" alt="Screenshot 2022-12-18 at 3 05 44 PM" src="https://user-images.githubusercontent.com/9396084/208293724-d1205da6-6546-4d0b-8e2c-dfded0e8553b.png"> |
| <img width="1512" alt="Screenshot 2022-12-18 at 3 10 10 PM" src="https://user-images.githubusercontent.com/9396084/208293728-8654ab6b-99eb-4e0b-a97a-c3caf23decb6.png"> | <img width="1512" alt="Screenshot 2022-12-18 at 3 27 51 PM" src="https://user-images.githubusercontent.com/9396084/208293734-cee8469d-7c16-46dd-8e3c-72e7155ea0af.png"> | 
| <img width="1512" alt="Screenshot 2022-12-18 at 3 41 29 PM" src="https://user-images.githubusercontent.com/9396084/208293740-f07aef76-0c44-408f-81be-90a1a9718a8d.png"> | <img width="1512" alt="Screenshot 2022-12-18 at 3 47 06 PM" src="https://user-images.githubusercontent.com/9396084/208293746-d0a65205-b817-4a5b-85a4-cf3cb33e8d5a.png"> |

There is one special case in pages, i.e. it was using `row` class from bootstrap which has code 
```

element.style {
}
.row {
    display: flex; 
    flex-wrap: wrap;
    margin-right: -15px;
    margin-left: -15px;
}
```

This was replaced with
`flex flex-wrap -mx-4` which gave similar results as shown below

| Before  | After |
| ------------- | ------------- |
| <img width="499" alt="Screenshot 2022-12-18 at 3 53 57 PM" src="https://user-images.githubusercontent.com/9396084/208293905-cba5db98-9718-4009-a5bf-75aba40ce8c2.png"> | <img width="501" alt="Screenshot 2022-12-18 at 3 58 12 PM" src="https://user-images.githubusercontent.com/9396084/208293912-9a32fe02-b691-4af3-a421-8641d8fd7117.png"> |

### UI accessibility concerns?

No

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
